src/main/kotlin/org/ionproject/integration/builder/implementations/IselTimetableTeachersBuilder.kt

```diff
@@ -0,0 +1,251 @@
+    }
+
+    private fun setCommonData(data: String, timetable: Timetable, courseTeacher: CourseTeacher) {
+        val school = RegexUtils.findMatches(SCHOOL_REGEX, data)[0].trimEnd()
+        val programme = RegexUtils.findMatches(PROGRAMME_REGEX, data, RegexOption.MULTILINE)[0].trimEnd()
+        val calendarTerm = RegexUtils.findMatches(CALENDAR_TERM_REGEX, data, RegexOption.MULTILINE)[0].replace("Ano Letivo:", "").trim()
+        val classSection = RegexUtils.findMatches(CLASS_SECTION_REGEX, data, RegexOption.MULTILINE)[0].replace("Turma:", "").trim()
+
+        timetable.school = school
+        timetable.programme = programme
+        timetable.calendarTerm = calendarTerm
+        timetable.classSection = classSection
+
+        courseTeacher.school = school
+        courseTeacher.programme = programme
+        courseTeacher.calendarTerm = calendarTerm
+        courseTeacher.classSection = classSection
+    }
+
+    private fun getCourseList(data: Array<Array<Cell>>): Try<List<Course>> {
+        var courseList = mutableListOf<Course>()
+        var weekdays = mutableMapOf<Double, String>()
+        var courseDetails: Triple<String, String, String>
+
+        return Try.of {
+            for (i in 0 until data.count()) {
+                val cells = data[i]
+
+                if (weekdays.keys.isEmpty()) {
+                    populateWeekdays(cells, weekdays)
+                    continue
+                }
+
+                var beginTime = LocalTime.now()
+
+                for (j in 0 until cells.count()) {
+                    val cell = cells[j]
+
+                    if (cell.text.isNullOrEmpty()) continue
+
+                    val matches = RegexUtils.findMatches(TIME_SLOT_REGEX, cells[j].text)
+                    if (matches.count() != 0) {
+                        beginTime = getBeginTime(matches)
+                        continue
+                    }
+
+                    cell.text.split('\r')
+                        .forEach {
+                            courseDetails = populateCourseDetails(it)
+
+                            var duration: Duration = if (cell.height > HEIGHT_THRESHOLD) {
+                                Duration.ofHours(3)
+                            } else {
+                                Duration.ofHours(1).plusMinutes(30)
+                            }
+
+                            courseList.add(
+                                Course(
+                                    acronym = courseDetails.first.trim(),
+                                    type = courseDetails.second,
+                                    room = courseDetails.third,
+                                    begin_time = beginTime.toString(),
+                                    end_time = beginTime.plusSeconds(duration.toSeconds()).toString(),
+                                    duration = duration.toString(),
+                                    weekday = weekdays.getOrDefault(cell.left, "")
```

Comment:
I think this should either return a Try.ofError and not be enclosed in a Try.of, or throw an exception if it remains inside it.